runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -1129,7 +1129,9 @@ Java_jdk_internal_misc_Unsafe_registerNatives(JNIEnv *env, jclass clazz)
 jboolean
  memOverlapIsNone(j9object_t sourceObject, UDATA sourceOffset, j9object_t destObject, UDATA destOffset, UDATA actualCopySize) {
 	jboolean result = JNI_FALSE;
-	if ((sourceObject == NULL) && (destObject == NULL)) {
+	if (sourceObject != destObject) {
+		result = JNI_TRUE;
+	} else if ((sourceObject == NULL) && (destObject == NULL)) {
```

Comment:
This is a conservative approach in that it leaves out the case where the objects are the same but the copied regions don't overlap.

I'm ok with this approach for JDK15 but we might want to dig further into whether it should be made less conservative.

Reply:
Created https://github.com/eclipse/openj9/issues/10560.